crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -1,226 +1,43 @@
-        pbr_input.material.flags = pbr_bindings::material.flags;
-        pbr_input.material.alpha_cutoff = pbr_bindings::material.alpha_cutoff;
-        pbr_input.frag_coord = in.position;
-        pbr_input.world_position = in.world_position;
-        pbr_input.is_orthographic = is_orthographic;
-        pbr_input.flags = mesh[in.instance_index].flags;
-
-        // emmissive
-        // TODO use .a for exposure compensation in HDR
-        var emissive: vec4<f32> = pbr_bindings::material.emissive;
-#ifdef VERTEX_UVS
-        if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_EMISSIVE_TEXTURE_BIT) != 0u) {
-            emissive = vec4<f32>(emissive.rgb * textureSampleBias(pbr_bindings::emissive_texture, pbr_bindings::emissive_sampler, uv, view.mip_bias).rgb, 1.0);
-        }
-#endif
-        pbr_input.material.emissive = emissive;
-
-        // metallic and perceptual roughness
-        var metallic: f32 = pbr_bindings::material.metallic;
-        var perceptual_roughness: f32 = pbr_bindings::material.perceptual_roughness;
-#ifdef VERTEX_UVS
-        if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_METALLIC_ROUGHNESS_TEXTURE_BIT) != 0u) {
-            let metallic_roughness = textureSampleBias(pbr_bindings::metallic_roughness_texture, pbr_bindings::metallic_roughness_sampler, uv, view.mip_bias);
-            // Sampling from GLTF standard channels for now
-            metallic = metallic * metallic_roughness.b;
-            perceptual_roughness = perceptual_roughness * metallic_roughness.g;
-        }
-#endif
-        pbr_input.material.metallic = metallic;
-        pbr_input.material.perceptual_roughness = perceptual_roughness;
-
-        // occlusion
-        // TODO: Split into diffuse/specular occlusion?
-        var occlusion: vec3<f32> = vec3(1.0);
-#ifdef VERTEX_UVS
-        if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_OCCLUSION_TEXTURE_BIT) != 0u) {
-            occlusion = vec3(textureSampleBias(pbr_bindings::occlusion_texture, pbr_bindings::occlusion_sampler, uv, view.mip_bias).r);
-        }
-#endif
-#ifdef SCREEN_SPACE_AMBIENT_OCCLUSION
-        let ssao = textureLoad(screen_space_ambient_occlusion_texture, vec2<i32>(in.position.xy), 0i).r;
-        let ssao_multibounce = gtao_multibounce(ssao, unlit_color.rgb);
-        occlusion = min(occlusion, ssao_multibounce);
-#endif
-        pbr_input.occlusion = occlusion;
+    // generate a PbrInput struct from the StandardMaterial bindings
+    var pbr_input = pbr_input_from_standard_material(in, is_front);
 
-        // world normal
-        pbr_input.world_normal = pbr_functions::prepare_world_normal(
-            in.world_normal,
-            (pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_DOUBLE_SIDED_BIT) != 0u,
-            is_front,
-        );
+    // alpha discard
+    pbr_input.material.base_color = alpha_discard(pbr_input.material, pbr_input.material.base_color);
```

Comment:
robtfm pointed out that this is not a regression. I am glad that when code moves around, I see it with fresh eyes. And, as this is not a regression, this is not a blocking comment.

Reply:
as we discussed, it's a little involved to calculate the gradient and include the mip_bias correctly, so i'd prefer to leave it for future

Reply:
This may not be a regression here, but before https://github.com/bevyengine/bevy/pull/10105 the discard happened much earlier https://github.com/bevyengine/bevy/blob/a15d1526358a3e5fd0ac76339f0c23612842167b/crates/bevy_pbr/src/render/pbr_functions.wgsl#L155 But in https://github.com/bevyengine/bevy/pull/10105 it was also mentioned "it must be done after all texture samples for uniform control flow"

Reply:
the discard was previously in `pbr_functions::pbr()` which was well after all the texture samples (which are in `pbr::fragment()`)